pkg/repository/v1_repository.go

```diff
@@ -54,6 +65,87 @@ func NewV1Repo(mirror Mirror, opts Options, local v1manifest.LocalManifests) *V1
 const maxTimeStampSize uint = 1024
 const maxRootSize uint = 1024 * 1024
 
+// UpdateComponents updates the components described by specs.
+// TODO tests
+func (r *V1Repository) UpdateComponents(specs []ComponentSpec) error {
+	err := r.ensureManifests()
+	if err != nil {
+		return err
+	}
+	for _, spec := range specs {
+		manifest, err := r.updateComponentManifest(spec.ID)
```

Comment:
`updateComponentManifest` assume we can get a higher version manifest from the mirror than the local one,
seems we can not call this directly

Reply:
Good catch!